core/src/dird/ua_restore.cc

```diff
@@ -839,15 +845,65 @@ static bool get_date(UaContext* ua, char* date, int date_len)
   ua->SendMsg(
       _("The restored files will the most current backup\n"
         "BEFORE the date you specify below.\n\n"));
+  std::string cmpdate;
   for (;;) {
     if (!GetCmd(ua, _("Enter date as YYYY-MM-DD HH:MM:SS :"))) { return false; }
-    if (StrToUtime(ua->cmd) != 0) { break; }
+    cmpdate = CompensateShortDate(ua->cmd);
+
+    if (StrToUtime(cmpdate.c_str()) != 0) { break; }
     ua->ErrorMsg(_("Improper date format.\n"));
   }
-  bstrncpy(date, ua->cmd, date_len);
+  bstrncpy(date, cmpdate.c_str(), date_len);
   return true;
 }
 
+/**
+ * Compensate missing date-time elements in shorter formats
+ * Returns a full compensated date when argument has a correct short format
+ * Returns the argument unchanged, if there is a format problem
+ */
+std::string CompensateShortDate(const char* cmd)
+{
+  tm datetime{};
+  char trailinggarbage[16]{""};
+
+  if ((sscanf(cmd, "%u-%u-%u %u:%u:%u%15s", &datetime.tm_year, &datetime.tm_mon,
+              &datetime.tm_mday, &datetime.tm_hour, &datetime.tm_min,
+              &datetime.tm_sec, &trailinggarbage)
+           == 7
+       || sscanf(cmd, "%u-%u-%u %u:%u%15s", &datetime.tm_year, &datetime.tm_mon,
+                 &datetime.tm_mday, &datetime.tm_hour, &datetime.tm_min,
+                 &trailinggarbage)
+              == 6
+       || sscanf(cmd, "%u-%u-%u %u%15s", &datetime.tm_year, &datetime.tm_mon,
+                 &datetime.tm_mday, &datetime.tm_hour, &trailinggarbage)
+              == 5
+       || sscanf(cmd, "%u-%u-%u%15s", &datetime.tm_year, &datetime.tm_mon,
+                 &datetime.tm_mday, &trailinggarbage)
+              == 4
+       || sscanf(cmd, "%u-%u%15s", &datetime.tm_year, &datetime.tm_mon,
+                 &trailinggarbage)
+              == 3
+       || sscanf(cmd, "%u%s", &datetime.tm_year, &trailinggarbage) == 2)
+      && (trailinggarbage[0] == '\0')) {
```

Comment:
For strings, you need to pass the array to `sscanf()` and not the address of the array.
```suggestion
  if ((sscanf(cmd, "%u-%u-%u %u:%u:%u%15s", &datetime.tm_year, &datetime.tm_mon,
              &datetime.tm_mday, &datetime.tm_hour, &datetime.tm_min,
              &datetime.tm_sec, trailinggarbage)
           == 7
       || sscanf(cmd, "%u-%u-%u %u:%u%15s", &datetime.tm_year, &datetime.tm_mon,
                 &datetime.tm_mday, &datetime.tm_hour, &datetime.tm_min,
                 trailinggarbage)
              == 6
       || sscanf(cmd, "%u-%u-%u %u%15s", &datetime.tm_year, &datetime.tm_mon,
                 &datetime.tm_mday, &datetime.tm_hour, trailinggarbage)
              == 5
       || sscanf(cmd, "%u-%u-%u%15s", &datetime.tm_year, &datetime.tm_mon,
                 &datetime.tm_mday, trailinggarbage)
              == 4
       || sscanf(cmd, "%u-%u%15s", &datetime.tm_year, &datetime.tm_mon,
                 trailinggarbage)
              == 3
       || sscanf(cmd, "%u%s", &datetime.tm_year, trailinggarbage) == 2)
      && (trailinggarbage[0] == '\0')) {
```